analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/NullPointerDereferenceBase.cs

```diff
@@ -44,20 +44,22 @@ protected override ProgramState PreProcessSimple(SymbolicContext context)
         operation.Kind switch
         {
             OperationKindEx.Invocation => NullInstanceCandidate(operation.ToInvocation()),
+            OperationKindEx.FieldReference => operation.ToFieldReference().Instance,
             OperationKindEx.PropertyReference => NullInstanceCandidate(operation.ToPropertyReference()),
+            OperationKindEx.EventReference => operation.ToEventReference().Instance,
             OperationKindEx.Await => operation.ToAwait().Operation,
             OperationKindEx.ArrayElementReference => operation.ToArrayElementReference().ArrayReference,
             _ => null,
         };
 
     private static IOperation NullInstanceCandidate(IInvocationOperationWrapper operation) =>
-        operation.TargetMethod.ContainingType.Is(KnownType.System_Nullable_T)
+        operation.TargetMethod.ContainingType.IsNullableValueType()
         && operation.TargetMethod.Name != nameof(Nullable<int>.GetType) // All methods on Nullable but .GetType() are safe to call
```

Comment:
You need to use the language specific naming comparer here. Add a test case for VB where GetType is written with a different casing.

Reply:
I tried to create a failing test, but it succeeded:
https://github.com/SonarSource/sonar-dotnet/pull/6900/files#diff-246113cb3c64b6c71878db844e581a6fe07ddae9f676fa1a0d1e0264e5dbec2aR101

Reply:
👍 The TargetMethod is the method symbol. It comes from the metadata from the framework and is therefore always "GetType".